packages/serverpod/lib/src/database/table_relation.dart

```diff
@@ -5,41 +5,97 @@ import 'package:serverpod/src/database/columns.dart';
 /// This is typically only used internally by the serverpod framework.
 @internal
 class TableRelation {
-  /// Name of table.
-  final String tableName;
+  /// Records the relationship between multiple tables.
+  /// Order is important, as it determines the order of joins in the query.
+  final List<TableRelationEntry> _tableRelationEntries;
 
-  /// Foreign table relation column.
-  final Column foreignTableColumn;
+  /// Creates a new [TableRelation].
+  /// Throws [ArgumentError] if [tableRelationEntries] is empty.
+  TableRelation(this._tableRelationEntries) {
+    if (_tableRelationEntries.isEmpty) {
+      throw ArgumentError('TableRelation must have at least one entry.');
+    }
+  }
 
-  /// Relation column.
-  final String column;
+  /// Builds all table relations required to join the tables.
+  List<TableRelation> get getRelations {
+    List<TableRelation> relations = [];
+    for (var i = 0; i < _tableRelationEntries.length; i++) {
+      relations.add(TableRelation(_tableRelationEntries.sublist(0, i + 1)));
+    }
+
+    return relations;
```

Comment:
Since table relation now is responsibly for building its own query alias, this method creates all table relations that can be built from this chain.

Think of a `TableRelation` that has `TableRelationEntries` for the table relations  A -> B -> C

When interacting with the `TableRelation` it can give us the necessary information to make a connection from B -> C.

But when performing the joins we also need information from A -> B to complete the link. That is the responsibility of this method. Creating a `TableRelation` for each step of the way that can give us the necessary information to make that link.